src/operators.jl

```diff
@@ -442,3 +442,83 @@ This is equivalent to [`induced_subgraph`](@ref)`(g, neighborhood(g, v, d, dir=d
 with respect to `v` (i.e. `:in` or `:out`).
 """
 egonet(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = g[neighborhood(g, v, d, dir=dir)]
+
+
+"""
+    merge_vertices!(g, vs)
+
+Combine vertices specified in `vs` into single vertex whose
+index will be the lowest value in `vs`.
+
+Supports Graphs and DiGraphs.
+
+All edges running to vertices in `vs` connect to new
+merged vertex.
+
+Return a vector with new vertex values index by original vertex
+indices.
+"""
+
+function merge_vertices!(g::AbstractGraph, vs::Vector{T} where T <: Integer)
+    vs = sort!(unique(vs))
+    merged_vertex = vs[1]
+    vs = vs[2:end]
+
+    x = zeros(Int, nv(g))
+    x[vs] = 1
+    new_vertex_ids = collect(1:nv(g)) .- cumsum(x)
+    new_vertex_ids[vs] = merged_vertex
+
+    for i in 1:nv(g)
+
+        # Adjust connections to merged vertices
+        if (i != merged_vertex) & !(i in vs)
+            nbrs_to_rewire = Set{eltype(g)}()
+            for j in g.fadjlist[i]
+               if j in vs
```

Comment:
I don't think that works here since it's a full `if`/`else` for `j in vs`, not just a filter.